src/client/src/components/homepage.js

```diff
@@ -0,0 +1,75 @@
+const React = require('react')
+const {Link, Redirect} = require('react-router-dom')
+const fetch = require('node-fetch')
+const logo = require('../logo.svg')
+const responseCodes = require('../statusCodeResponses')
+const LOGIN_URL = '/login'
+
+class HomePage extends React.Component {
+    constructor(props){
+        super(props)
+
+        this.state = {
+            username: '',
+            password: '',
+            successfulLogin: false
+        }
+
+        this.handleChange = this.handleChange.bind(this)
+        this.handleSubmit = this.handleSubmit.bind(this)
+    }
+
+    loginHandler(username, password){
+        fetch(LOGIN_URL, {
+            method: 'POST',
+            headers: { 'Content-Type': 'application/json' },
+            body: JSON.stringify({username, password}),
+            credentials: 'same-origin'
+        }).then(response => {
+            if(response.status === 200){
+                this.setState({successfulLogin: true})
+            }
+            if(responseCodes.hasOwnProperty(response.status)){
+                alert(responseCodes[response.status])
+                console.log(response.status, response)
+            }
+        }).catch(err => console.err(err))
+    }
+
+    handleChange(event){
+        const settings = {'Username': 'username', 'Password': 'password'}
+        const key = settings[event.target.placeholder]
+        this.setState({ [key] : event.target.value})
```

Comment:
I don't really understand, what are you performing here.

Reply:
I'm abusing the placeholder property of a component to reduce needed amount of code to create dynamic object keys for state changes.

Reply:
Done